sdk/webpubsub/azure-messaging-webpubsubclient/azure/messaging/webpubsubclient/aio/_client.py

```diff
@@ -0,0 +1,1179 @@
+                _LOGGER.debug("fail to reconnect, and will retry in another task")
+                break
+            except Exception as e:  # pylint: disable=broad-except
+                _LOGGER.warning("An attempt to reconnect connection failed: %s", e)
+        if not success:
+            await self._handle_connection_stopped()
+        else:
+            _LOGGER.debug("reconnect successfully")
+
+    async def _handle_connection_stopped(self):
+        _LOGGER.debug("Connection stopped")
+        self._is_stopping = False
+        self._state = WebPubSubClientState.STOPPED
+        self._call_back(CallbackType.STOPPED)
+
+    async def _handle_connection_close_and_no_recovery(self, reconnect_tried_times: Optional[int] = None):
+        _LOGGER.debug("Connection closed and no recovery")
+        self._state = WebPubSubClientState.DISCONNECTED
+        self._call_back(
+            CallbackType.DISCONNECTED,
+            OnDisconnectedArgs(
+                connection_id=self._connection_id,
+                message=(self._last_disconnected_message.message if self._last_disconnected_message else None),
+            ),
+        )
+        if self._auto_reconnect:
+            await self._handle_auto_reconnect(reconnect_tried_times)
+        else:
+            await self._handle_connection_stopped()
+
+    def _is_connected(self) -> bool:
+        """check whether the client is still connected to server after open
+
+        :return: True if the client is connected to server, otherwise False
+        :rtype: bool
+        """
+        return self._state == WebPubSubClientState.CONNECTED and self._ws and not self._ws.closed  # type: ignore
+
+    async def _rejoin_group(self, group_name: str):
+        async def _rejoin_group_func():
+            try:
+                await self._join_group_core(group_name)
+                _LOGGER.debug("rejoin group %s successfully", group_name)
+            except Exception as e:  # pylint: disable=broad-except
+                _LOGGER.debug("fail to rejoin group %s", group_name)
+                self._call_back(
+                    CallbackType.REJOIN_GROUP_FAILED,
+                    OnRejoinGroupFailedArgs(group=group_name, error=e),
+                )
+
+        await _rejoin_group_func()
+
+    async def _connect(
+        self,
+        url: str,
+        reconnect_tried_times: Optional[int] = None,
+        recover_start_time: Optional[float] = None,
+    ):  # pylint: disable=too-many-statements
+        async def on_open():
+            if self._is_stopping:
+                try:
+                    if self._ws:
+                        await self._ws.close()
+                finally:
+                    raise OpenClientError("Can't open a client during stopping")
+
+            _LOGGER.debug("WebSocket connection has opened")
+            self._state = WebPubSubClientState.CONNECTED
+
+        async def on_message(data: str):
+            async def handle_ack_message(message: AckMessage):
```

Comment:
Same comment as above - not sure why we are making nested functions that we just immediately call rather than simply putting this logic into the switch statement below?

Reply:
Just personal thought: putting the logic with switch statement will make the switch logic too many lines.

Reply:
![image](https://github.com/Azure/azure-sdk-for-python/assets/70930885/c4d28fb4-3c4d-4f10-8455-a2f8571175cd)

Reply:
We typically want to avoid nested functions, more so because there is a perf implication of calling functions as well. Perf being a key component for this SDK 

As it is the method is long, so moving it out into the switch wont impact the line of code overall and have us avoiding function calls.

Reply:
I see.
Updated: I cancel nested function and put the code directly into swtich-statement for Async and Sync.